api_tests/preprints/views/test_preprint_list_mixin.py

```diff
@@ -73,24 +73,59 @@ def test_filter_published_false_public(self):
         res = self.app.get('{}filter[is_published]=false'.format(self.url))
         assert len(res.json['data']) == 0
 
+@pytest.mark.django_db
 class PreprintIsValidListMixin(object):
 
-    def setUp(self):
-        super(PreprintIsValidListMixin, self).setUp()
-        assert self.admin, 'Subclasses of PreprintIsValidListMixin must define self.admin'
-        assert self.project, 'Subclasses of PreprintIsValidListMixin must define self.project'
-        assert self.provider, 'Subclasses of PreprintIsValidListMixin must define self.provider'
-        assert self.url, 'Subclasses of PreprintIsValidListMixin must define self.url'
+    # FIXTURES
+    @pytest.fixture()
+    def admin(self):
+        raise NotImplementedError("subclass must define an admin fixture")
+
+    @pytest.fixture()
+    def url(self):
+        raise NotImplementedError("subclass must define a url fixture")
+
+    @pytest.fixture()
+    def project(self, admin):
+        raise NotImplementedError("subclass must define a project fixture")
+
+    @pytest.fixture()
+    def provider(self):
+        raise NotImplementedError("subclass must define a provider")
+
+    @pytest.fixture()
+    def app(self):
+        return JSONAPITestApp()
+
+    @pytest.fixture()
+    def write_contrib(self):
+        return AuthUserFactory()
+
+    @pytest.fixture()
+    def non_contrib(self):
+        return AuthUserFactory()
+
+    @pytest.fixture()
+    def subject(self):
+        return SubjectFactory()
+
+    # SETUP
+    @pytest.fixture(scope="function", autouse=True)
+    def int(self, app, url, admin, write_contrib, non_contrib, project, provider, subject):
```

Comment:
`int` is a built-in for integers, probably just name this setUp like in the previous style